java/src/main/java/genericpubsub/ManagedSubscribe.java

```diff
@@ -230,50 +234,30 @@ public synchronized void close() {
      */
     private synchronized void abort(Exception e) {
         serverStream.onError(e);
-        isActive = false;
+        isActive.set(false);
         this.notifyAll();
     }
 
     /**
-     *  function illustrates the statues of the stream on a fixed intervals.
-     *  This is for demonstration purposes only, and not to be used in a production setting.
+     * Helper function to halt the current thread.
      */
-    public synchronized boolean waitForEvents(int numEvents) {
-        return waitForEventsWithMaxTimeOut(numEvents, Integer.MAX_VALUE);
-    }
-
-    public synchronized boolean waitForEventsWithMaxTimeOut(int numEvents, int timeoutInSeconds) {
-        final Instant startTime = Instant.now();
-        while (isActive && receivedEvents < numEvents) {
-            final long elapsed = Duration.between(startTime, Instant.now()).getSeconds();
-
-            if (elapsed > timeoutInSeconds) {
-                isActive = false;
-                logger.warn("Waiting for events for more than {} seconds", timeoutInSeconds);
-                return receivedEvents < numEvents;
-            }
-
+    public void waitInMillis(long duration) {
+        synchronized (this) {
             try {
-                logger.info("received {} out of {}", receivedEvents, numEvents);
-                this.wait(10_000);
+                this.wait(duration);
             } catch (InterruptedException e) {
-                logger.warn("interrupted while waiting", e);
-                break;
+                throw new RuntimeException(e);
             }
         }
-        logger.info("Total Received Events {} out of {}", receivedEvents, numEvents);
-        return receivedEvents < numEvents;
     }
 
-
     public static void main(String args[]) throws IOException  {
         ExampleConfigurations exampleConfigurations = new ExampleConfigurations("arguments.yaml");
 
         // Using the try-with-resource statement. The CommonContext class implements AutoCloseable in
         // order to close the resources used.
         try (ManagedSubscribe subscribe = new ManagedSubscribe(exampleConfigurations)) {
             subscribe.startManagedSubscription();
-            subscribe.waitForEvents(Math.min(5, exampleConfigurations.getNumberOfEventsToSubscribeInEachFetchRequest()));
```

Comment:
Removing this should allow the subscriber to continue indefinitely (as opposed to automatically closing once a certain number of events are received)